### ISSUE TITLE
Fix Cohesion Problem

### ISSUE THREAD

**Issue by Victor-Crawshaw** [CMU-17-214/s2024#12]
FormattedTextDocument appears to be a god class, as there are 3 other interfaces: FormattedTextContent, Paragraph, and TextFragment within this class, as long as many other static methods. This class is doing far too much and needs to be refactored.

**Comment by Victor-Crawshaw** [comment]
I fixed the issue by separating each interface into its own file along with their accompanying static classes/records that implement them. I also updated all instances of these objects in the WebGen files and Project Parser. https://github.com/CMU-17-214/s24-hw4-webgen-Victor-Crawshaw/commit/d4ec80e6ebc4ce77fe30abb1b337af226b77ad30

**Comment by Victor-Crawshaw** [comment]
I fixed the issue by separating each interface into its own file along with their accompanying static classes/records that implement them. Now, FormattedTextDocument and these new classes are not God classes and doing too much. The project is much more cohesive this way. I also updated all instances of these objects in the WebGen files and Project Parser. https://github.com/CMU-17-214/s24-hw4-webgen-Victor-Crawshaw/commit/d4ec80e6ebc4ce77fe30abb1b337af226b77ad30
